### PR TITLE
Move KIC docs to kic-v2 folder

### PR DESCRIPTION
**Pull request by mheap**
### Description

Renamed `_src/kubernetes-ingress-controller` to `_src/kic-v2` in preparation for KIC v3 docs

### Testing instructions

Preview link: /kubernetes/ingress-controller/

### Checklist 

- [x] Review label added 
- [x] PR pointed to correct branch (`main` for immediate publishing, or a release branch: e.g. `release/gateway-3.2`, `release/deck-1.17`)

### CODE REVIEW

File: app/_data/docs_nav_kic_2.10.x.yml [hunk 1]
```diff
@@ -4,163 +4,229 @@ generate: true
...
+        src: /kic-v2/faq
       - text: Version Support Policy
         url: /support-policy
+        src: /kic-v2/support-policy
       - text: Stages of Software Availability
         url: /availability-stages
+        src: /kic-v2/availability-stages
       - text: Changelog
-        url: https://github.com/Kong/kubernetes-ingress-controller/blob/main/CHANGELOG.md
+        url: >-
```
Comment (lena-larionova) [inline app/_data/docs_nav_kic_2.10.x.yml:20]: Why does this get added?

Comment (mheap) [reply app/_data/docs_nav_kic_2.10.x.yml:20]: The YAML parser thinks that the URL is too long for a single line

Comment (lena-larionova) [reply app/_data/docs_nav_kic_2.10.x.yml:20]: I guess it doesn't seem to hurt anything, the output works fine.

**Review by lena-larionova — APPROVED** [review]
LGTM! Spot checked the output, didn't run into anything broken.

File: app/_data/docs_nav_kic_2.10.x.yml (hunk 1 above)
Comment (mheap) [reply app/_data/docs_nav_kic_2.10.x.yml:20]: Yep, it's valid YAML
CleanShot 2023-10-06 at 17 28 35

**Pull request merged**